### ISSUE TITLE
Problem with ExJsx dependency

### ISSUE THREAD

**Issue by julien-leclercq** [parroty/exvcr#153]
Hello ! 

Would you be encline to accept a patch to allow usage of json toolkit other than `exjsx` ? 

In several repo where I work (@fewlinesco) we have rabbitmq clients dependencies blocked because it requires a later version of `jsx` which is unable to install because `exjsx` has a too early dependency on `jsx`. I am asking you and not `exjsx` because there was no commits for 3 years on `exjsx`. 

What I plan to do is forking `exvcr` and make the stuff so it can use `Jason` for exemple instead of `exjsx`. This would allow us to get rid of this outdated dependency. If you are okay with that, I will provide the PR.

**Comment by tverlaan** [comment]
Hi @julien-leclercq , I'm running into the same issue but I see your PR was never merged and you closed it. What did you do to solve it? (or perhaps you didn't?)

To the owner, what kind of PR would you accept for this issue?

I'd appreciate any help, thanks :)

**Comment by thbar** [comment]
I am not the owner, but met the same issue. I'll add more information:
- `exjsx` is today marked as `This repository has been archived by the owner. It is now read-only.`.
- `ex_aws` (commonly used) enforces `jsx ~> 3.0`, where `exjsx` is enforcing `jsx -> 2.8.0`, so having `ExVCR` onboard now means one cannot upgrade to latest `ex_aws`.

```
Failed to use "jsx" because
  ex_aws (versions 2.2.0 to 2.2.3) requires ~> 3.0
  exjsx (version 4.0.0) requires ~> 2.8.0
```

So at this point using `exvcr` in a project blocks the upgrade of a production library.

Happy to help if possible!

**Comment by thbar** [comment]
I wonder if one path could be to "just" stop using `exjsx` and instead use `jsx` directly (or alternatively, use a configurable back-end like started in #154).

**Comment by swissbeats93** [comment]
I'm having the same problem. Is there any hope that this will be updated?

**Comment by jherdman** [comment]
I started #179 this morning. I wouldn't mind a little help with some of the test failures.

**Comment by fmmatheus** [comment]
Also having the same problem. Any updates?

**Comment by jherdman** [comment]
Nope! I'm a bit stuck on what to do here, and have had a lack of time to resolve the remaining issue. As far as I can tell the remaining failure requires some modification to how we record headers. I believe they need to be encoded using something like base 64 before we write them to a String. Give the PR a whirl. I'm happy to have some help 😄

**Comment by thbar** [comment]
FWIW, `ex_aws` has relaxed its dependency on JSX, reducing the trouble a bit for us (and maybe for others!) https://github.com/ex-aws/ex_aws/pull/827

**Comment by ffloyd** [comment]
Any update here? Rabbit libraries focus on 3.0 jsx version so I cannot use VCR with rabbit. =(
